### ISSUE TITLE
useGraphqlForms payload not updating on user data entry after first update

### ISSUE THREAD

**Issue by ACoolTurtle** [tinacms/tina-graphql-gateway#308]
## Description
On very complex data structures when using the `useGraphqlForms` hook, the payload component does not update after the first character has been entered by the user. It appears the be an issue of nested objects, but I may be wrong.

### Steps to reproduce

Create a fairly complex JSON object and query it from Tina.  Check how often the data is updated in conjunction to user data entry.

```tsx

const [payload, isLoading] = useGraphqlForms<Query>({
	query,
	variables: {
		relativePath: `${slug}.json`,
	},
	formify: ({ createForm, formConfig }) => {
		return createForm(formConfig);
	},
});

useEffect(() => {
	console.log("PAYLOAD", payload?.getPageDocument?.data);
}, [JSON.stringify(payload?.getPageDocument?.data)]);
```

Example JSON to be queried: 

```json
{
	"slug": "/",
	"_template": "page",
	"seo": {
		"title": "Home",
		"description": "Description",
		"image": "/some/image",
		"lang": "en",
	},
	"menuLinks": [
		{
			"title": "Home",
			"path": "/"
		},
	],
	"components": [
		{
			"template": "carousel",
			"images": [
				{
					"image": "/images/image.jpg"
				},
			],
			"height": 60,
			"fade": false,
			"fadeTime": 50,
			"fadeOnce": true,
			"delay": 300,
			"toggle_fade_direction": false
		},
		{
			"template": "title",
			"titleText": "Title",
			"fade": true,
			"fadeTime": 50,
			"fadeOnce": true,
			"delay": 300,
			"mdx": "### MDX String",
			"toggle_fade": false,
			"toggle_fade_direction": false
		},
		{
			"template": "link_grid",
			"fade": true,
			"fadeTime": 20,
			"fadeOnce": true,
			"delay": 300,
			"grid": [
				{
					"titleText": "Title",
					"description": "Description",
					"link": "URL",
					"image": "/images/image.jpg",
					"mdx": "* MDXString",
					"linkText": "TEXT"
				}
			],
			"toggle_fade_direction": false
		}
	]
}

```

### Expected result

Data to change on every data entry

### Actual result

No data change after the very first time the data has been entered. This only seems to come into effect when we are using multiple nested objects

### Environment

<html>
<body>
<h4 style="box-sizing: border-box; margin-bottom: 16px; margin-top: 24px; font-weight: 600; font-size: 14px; line-height: 1.25; color: rgb(201, 209, 217); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Tinacms Package Versions</h4><div style="box-sizing: border-box; color: rgb(201, 209, 217); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><pre style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Consolas, &quot;Liberation Mono&quot;, Menlo, monospace; font-size: 11.9px; margin-bottom: 16px; margin-top: 0px; overflow-wrap: normal; background-color: var(--color-bg-tertiary); border-radius: 6px; line-height: 1.45; overflow: auto; padding: 16px;"><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Consolas, &quot;Liberation Mono&quot;, Menlo, monospace; font-size: 11.9px; background: transparent; border-radius: 6px; margin: 0px; padding: 0px; border: 0px; white-space: pre; word-break: normal; overflow-wrap: normal; display: inline; line-height: inherit; overflow: visible;">
react-tinacms-editor@npm:0.41.1
react-tinacms-inline@npm:0.41.1
tinacms@npm:0.41.1
</code><p style="box-sizing: border-box; margin-bottom: 16px; margin-top: 0px;"><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Consolas, &quot;Liberation Mono&quot;, Menlo, monospace; font-size: 10.115px; background-color: transparent; border-radius: 6px; margin: 0px; padding: 0px; overflow-wrap: normal; border: 0px; display: inline; line-height: inherit; overflow: visible;"></code></p></pre></div><p style="box-sizing: border-box; margin-bottom: 16px; margin-top: 0px; color: rgb(201, 209, 217); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"></p><h4 style="box-sizing: border-box; margin-bottom: 16px; margin-top: 24px; font-weight: 600; font-size: 14px; line-height: 1.25; color: rgb(201, 209, 217); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Tools</h4>

Tool | Version
-- | --
browser | Chrome 91.0.4472.106
os | Windows 10, WSL 2 Ubuntu
node | 15.7.0
npm | 7.4.3

</body>
</html>

**Comment by logan-anderson** [comment]
@ACoolTurtle Thanks again for all the reports! Just a couple things we need to reproduce this error.

1. Your `schema.ts` (and all relevant files (if its the same as the other issue  I can get it from there) )
2. The graphql query you are using.

**Comment by ACoolTurtle** [comment]
@logan-anderson of course, it's the same query and schema as in #293. Sorry about all of the reports!

**Comment by logan-anderson** [comment]
@ACoolTurtle Don't be sorry about the reports! Thanks for finding them so we can fix them and make this product the best it can be.

Unfortunately I was not able to reproduce  your error. What field are you editing where you are getting this behaviour?

On a side note: We are continuing to make Tina Cloud and Tina open source better and if you have time to chat we would love to hear about how you are finding the overall DX (whats good, whats bad, how can we improve). 

Here is a calendar link if you want to book a 15-30min chat: https://calendly.com/logan-forestry/15min

**Comment by ACoolTurtle** [comment]
@logan-anderson It is honestly any field, the fields that come to mind are markdown and text fields though. This bug is very strange however, as I have 3 different page types and the bug is only present on one. My running theory is because of the multiple nested objects, but that's just a guess.

I would love to have a chat at some point, however I don't have much free time right now. Do you have a form I could fill out in the mean time?

**Comment by logan-anderson** [comment]
@ACoolTurtle Ill look into this more. You can go to this link and book directly into my calendar (pick anytime that works for you). https://calendly.com/logan-forestry/15min
